### PR TITLE
Event generation using test_mpc and IPA integration test + migrate to HTTP2

### PR DESCRIPTION
**Pull request by akoshelev**
There are there major changes that this PR brings:

* Implement event generator as `Iterator<TestRawDataRecord>`. for millions of events that is more efficient than the existing approach. It takes about 3 seconds to generate 1 million events (excluding IO)
* Add new `gen-ipa-inputs` command
* IPA integration test that uses `gen-ipa-inputs` to generate inputs and run semi-honest IPA.

--edit
Unintended change, but IPA integration tests were failing (see #650) on HTTP 1.1, so I had to include changing clients to use HTTP2

### CODE REVIEW

File: tests/helper_networks.rs
Comment (akoshelev) [inline tests/helper_networks.rs:224]: this is a bit annoying - I know that there is no guaranteed way to pick an available port (at least on MacOS, on Linux it works pretty reliable with SO_REUSEPORT and rust appears to be using [it](https://github.com/rust-lang-nursery/rust-cookbook/issues/500)), but even best-effort approach with TcpListener should work here. Right now it is not possible to test this in parallel

Comment (andyleiserson) [reply tests/helper_networks.rs:224]: I thought about this when I wrote the test... a possible strategy I came up with is to create `TcpListeners` in the test, pass them to the helper processes as /dev/stdin, and add an option to the helper that does `TcpListener::from_raw_fd` rather than bind a socket. I didn't do this at the time because it seemed like a fair amount of work, and it seemed useful to have at least one test that exercises the standard flow for binding the socket on startup. But if we're adding more tests that follow this framework, it becomes more important to do this work. I'll see if I can get this done next week.

**Comment by akoshelev** [comment]
integration tests hanging is not good: https://github.com/private-attribution/ipa/actions/runs/4964162687/jobs/8884058950?pr=647, opening an issue for that

File: src/bin/test_mpc.rs
Comment (martinthomson) [inline src/bin/test_mpc.rs:96]: the oneshot test uses -n (for number of records)

File: src/cli/csv.rs
Comment (martinthomson) [inline src/cli/csv.rs:3]: Do we name this simply `Serializer` (and then re-export as `CsvSerializer`?

File: src/test_fixture/event_gen.rs
Comment (martinthomson) [inline src/test_fixture/event_gen.rs:108]: Do we even need to track this?  I mean, the problem here is that we're essentially stateless, but you could create user IDs in monotonically increasing sequence.  A PRP would allow you to have the identifiers that you use appear to be randomly distributed in the target range...

File: src/test_fixture/event_gen.rs
Comment (martinthomson) [inline src/test_fixture/event_gen.rs:241]: I would test that timestamps are monotonic as well.

File: src/test_fixture/event_gen.rs
Comment (martinthomson) [inline src/test_fixture/event_gen.rs:188]: This seems a little small for larger event counts.  It's arbitrary, but I'd have said that you would expect hundreds of people for a million records.  (It's no extra work for IPA, but I'm not sure how much we care about fidelity.)

**Review by martinthomson — APPROVED** [review]
@akoshelev and @andyleiserson should work out who gets to merge and who get to rebase :)

**Comment by akoshelev** [comment]
> @akoshelev and @andyleiserson should work out who gets to merge and who get to rebase :)

@andyleiserson lets merge your changes first

**Comment by richajaindce** [comment]
> > @akoshelev and @andyleiserson should work out who gets to merge and who get to rebase :)
> 
> @andyleiserson lets merge your changes first

There are 2 PRs which would need to be rebased based on what gets landed first (#654 also) :P 
How about, we land this since it is already approved?

**Comment by andyleiserson** [comment]
> How about, we land this since it is already approved?

That's fine with me.

**Comment by akoshelev** [comment]
I am on vacation until Tuesday PT time, so I'll merge it to unblock you folks. Sorry @andyleiserson for the rebase

**Pull request merged**
